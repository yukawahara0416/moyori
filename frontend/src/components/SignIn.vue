<template>
  <div>
    <h2>SignIn</h2>
    <input type="text" v-model="signInFormData.email" placeholder="email" />
    <input
      type="text"
      v-model="signInFormData.password"
      placeholder="password"
    />
    <v-btn type="submit" @click="signIn">ログイン</v-btn>
    <v-btn @click="signInAsTestUser" type="submit">
      テストユーザでログイン
    </v-btn>
    <div>{{ signInFormData }}</div>
    <div>{{ currentUser }}</div>
    <v-btn v-show="currentUser" type="submit" @click="signOut">
      ログアウト
    </v-btn>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  data() {
    return {
      testUser: {
        email: 'tester@example.com',
        password: 'password'
      }
    }
  },

  computed: {
    ...mapGetters(['signInFormData', 'currentUser'])
  },

  methods: {
    signIn() {
      this.$store.dispatch('signIn', this.signInFormData)
    },
    signInAsTestUser() {
      this.$store.dispatch('signIn', this.testUser)
    },
    signOut() {
      this.$store.dispatch('signOut')
    }
  }
}
</script>
