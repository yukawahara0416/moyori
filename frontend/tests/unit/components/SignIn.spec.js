import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import SignIn from '@/components/SignIn.vue'
import userStore from '@/store/modules/user.js'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let store
let state
let actions

beforeEach(() => {
  state = {
    currentUser: {},
    signInFormData: {
      email: 'hoge@example.com',
      password: 'password'
    }
  }

  actions = {
    signIn: jest.fn()
  }

  store = new Vuex.Store({
    modules: {
      userStore: {
        state,
        getters: userStore.getters,
        actions
      }
    }
  })

  wrapper = shallowMount(SignIn, {
    localVue,
    store
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('v-on', () => {
  it('signIp', () => {
    const event = jest.fn()
    wrapper.setMethods({ signIn: event })
    wrapper.findAll('[data-test="signin"]').trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('getters', () => {
  it('signInFormData', () => {
    expect(wrapper.vm.signInFormData).toEqual(state.signInFormData)
  })

  it('currentUser', () => {})
})

describe('actions', () => {
  it('signIn', () => {
    wrapper.vm.signIn()
    expect(actions.signIn).toHaveBeenCalled()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
