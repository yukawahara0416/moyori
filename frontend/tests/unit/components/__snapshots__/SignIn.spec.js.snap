// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div>
  <h2>
    SignIn
  </h2>
   
  <input
    placeholder="email"
    type="text"
  />
   
  <input
    placeholder="password"
    type="text"
  />
   
  <button
    class="v-btn v-btn--contained theme--light v-size--default"
    type="submit"
  >
    <span
      class="v-btn__content"
    >
      ログイン
    </span>
  </button>
   
  <button
    class="v-btn v-btn--contained theme--light v-size--default"
    type="submit"
  >
    <span
      class="v-btn__content"
    >
      
    テストユーザでログイン
  
    </span>
  </button>
   
  <div>
    {
  "email": "hoge@example.com",
  "password": "password"
}
  </div>
   
  <button
    class="v-btn v-btn--contained theme--light v-size--default"
    type="submit"
  >
    <span
      class="v-btn__content"
    >
      
    ログアウト
  
    </span>
  </button>
</div>
`;
